CONTRIBUTING.md

```diff
@@ -34,4 +34,67 @@ Docs for `TODO(template) template_crate`:
 
 ```sh
 RUSTDOCFLAGS="--html-in-header katex-header.html" cargo doc --no-deps -p template_crate --open
-```
\ No newline at end of file
+```
+
+## Performance
+When contributing, besides correctness, it is also important to ensure good performance and reproducibility of the results.
+We recommend using [Criterion](https://crates.io/crates/criterion) for general benchmarking, as it provides a well-structured framework that allows reproducible benchmarks by just running a few commands. 
```

Comment:
For criterion description, I would add a few words regarding `--save-baseline` and `--baseline` arguments that aren't obvious for someone who hasn't dealt with criterion before.
They are super useful when iterating over your changes on top of existing reference implemenation (e.g. master branch)